# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: cdaa8a4e9e36
# Full Hash: cdaa8a4e9e366ad041b429a8f5bd0b9b259d4722
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 9ff26bcc580c -r cdaa8a4e9e36 dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Mon Jun 08 20:35:38 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Mon Jun 08 20:58:03 2020 +0000
@@ -723,8 +723,6 @@
   bool mIsCanvasDirty = false;
   uvec2 mRequestedSize = {};
 
-  RefPtr<gfx::DataSourceSurface> mFrontBufferSnapshot;
-
  public:
   explicit ClientWebGLContext(bool webgl2);
 
@@ -997,7 +995,8 @@
   void Present(WebGLFramebufferJS*, layers::TextureType);
   Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
                                                   layers::TextureType);
-  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
+  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot(
+      bool requireAlphaPremult = true) override;
 
  private:
   RefPtr<gfx::DataSourceSurface> BackBufferSnapshot();